openedx/features/badging/helpers.py

```diff
@@ -108,7 +112,8 @@ def filter_earned_badge_by_joined_team(user, course, earned_badges):
         flag: Has user joined any team
         earned_badges: All badges earned in a course, specific to joined team
     """
-    course_team = CourseTeam.objects.filter(course_id=course.id, users=user).values('team__room_id').first()
+    # course_team = CourseTeam.objects.filter(course_id=course.id, users=user).values('team__room_id').first()
+    course_team = CourseTeam.objects.filter(course_id=course.id, users=user).values('team_id', 'team__room_id').first()
```

Comment:
as we don't have `id` in `course_team`, So please update `error = FILTER_BADGES_ERROR.format(team_id=course_team['id'])` to avoid crash..

Reply:
As per above query following check seems extra which will never be True. 
```
if not course_team['team__room_id']:
        error = FILTER_BADGES_ERROR.format(team_id=course_team['id'])
        log.exception(error)
        raise Exception(error)
```

Reply:
I think this block of code should not be removed, It is written with intention of being on safe side. Team Group Chat has foreign key to Course Team but reverse is not true. Hence, need for this check

Reply:
`FILTER_BADGES_ERROR.format(team_id=course_team['id'])` is fixed